ui/push-health/Test.jsx

```diff
@@ -26,6 +48,102 @@ class Test extends PureComponent {
     }
   }
 
+  addSelectedTest = (test) => {
+    this.setState((prevState) => ({
+      selectedTests: prevState.selectedTests.add(test),
+    }));
+  };
+
+  removeSelectedTest = (test) => {
+    const { selectedTests } = this.state;
+    selectedTests.delete(test);
+    this.setState({
+      selectedTests,
+    });
+  };
+
+  retriggerSelected = (times) => {
+    const { notify, currentRepo, jobs } = this.props;
+    const { selectedTests } = this.state;
+
+    // Reduce down to the unique jobs
+    const testJobs = Array.from(selectedTests).reduce(
+      (acc, test) => ({
+        ...acc,
+        ...jobs[test.jobName].reduce((fjAcc, job) => ({ [job.id]: job }), {}),
+      }),
+      {},
+    );
+    const uniqueJobs = Object.values(testJobs);
+
+    JobModel.retrigger(uniqueJobs, currentRepo, notify, times);
+  };
+
+  markAsInvestigated = () => {
+    const { selectedTests } = this.state;
+    const { notify, currentRepo, revision, updatePushHealth } = this.props;
+
+    let data;
+    let failureStatus;
+    if (selectedTests.size === 0) {
+      notify(`Select atleast one test`, 'warning');
+    } else {
+      selectedTests.forEach(async (test) => {
+        ({ data, failureStatus } = await create(
+          `${getProjectUrl(
```

Comment:
It would be more efficient to save this url outside the for-loop so you don't create it each time you loop.